test/model/message_test.dart

```diff
@@ -52,16 +57,38 @@ void main() {
   /// Perform the initial message fetch for [messageList].
   ///
   /// The test case must have already called [prepare] to initialize the state.
+  ///
+  /// This does not support submessages. Use [prepareMessageWithSubmessage]
+  /// instead if needed.
   Future<void> prepareMessages(
     List<Message> messages, {
     bool foundOldest = false,
   }) async {
+    assert(messages.every((message) => message.poll == null));
     connection.prepare(json:
       newestResult(foundOldest: foundOldest, messages: messages).toJson());
     await messageList.fetchInitial();
     checkNotifiedOnce();
   }
 
+  /// Perform a single-message initial message fetch for [messageList] with
+  /// submessages.
+  ///
+  /// The test case must have already called [prepare] to initialize the state.
+  Future<void> prepareMessageWithSubmessage(
+    Message message, List<Submessage> submessages, {
```

Comment:
nit:
```suggestion
  Future<void> prepareMessageWithSubmessages(
    Message message, List<Submessage> submessages, {
```

since it takes a list of submessages, not just one